_posts/2021-04-20-custom-errors.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: post
+published: true
+title:  Custom Errors in Solidity
+date:   2021-04-20
+author: Solidity Team
+category: Explainers
+---
+
+[Solidity v0.8.4](https://github.com/ethereum/solidity/releases/tag/v0.8.4) provides a convenient
+and gas-efficient way to explain to users why an operation failed. Errors can be defined inside and
+outside of contracts (including interfaces and libraries).
+
+## Example
+
+The following contract shows an example usage of an Error:
+
+```solidity
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.4;
+
+contract VendingMachine {
+    address payable owner = payable(msg.sender);
+    error Unauthorized();
+    function withdraw() public {
+        if (msg.sender != owner)
+            revert Unauthorized();
+
+        owner.transfer(address(this).balance);
+    }
+    // ...
+}
+```
+
+The syntax of errors is similar to that of
+[events](https://docs.soliditylang.org/en/latest/contracts.html#events). They have to be used
+together with the [revert
+statement](https://docs.soliditylang.org/en/latest/control-structures.html#revert-statement) which
+causes all changes in the current call to be reverted and passes the error data back to the caller.
+
```

Comment:
Maybe add: Using errors together with ``require`` is not yet supported (see below)

and then further down explain that ``require(c, "abc")`` should be translated into ``if (!c) revert Abc();``